src/mlpack/bindings/cli/parse_command_line.hpp

```diff
@@ -34,27 +35,19 @@ PARAM_FLAG("version", "Display the version of mlpack.", "V");
  */
 void ParseCommandLine(int argc, char** argv)
 {
-  // First, we need to build the boost::program_options variables for parsing.
-  using namespace boost::program_options;
-  options_description desc;
-  variables_map vmap;
+  // First, we need to build the CLI11 variables for parsing.
+  CLI::App app;
+  app.set_help_flag();
 
   // Go through list of options in order to add them.
   std::map<std::string, util::ParamData>& parameters = CLI::Parameters();
-  typedef std::map<std::string, util::ParamData>::const_iterator IteratorType;
-  std::map<std::string, std::string> boostNameMap;
-  for (IteratorType it = parameters.begin(); it != parameters.end(); ++it)
+  using ItType = std::map<std::string, util::ParamData>::iterator;
+
+  for (ItType it = parameters.begin(); it != parameters.end(); ++it)
   {
     // Add the parameter to desc.
-    const util::ParamData& d = it->second;
-    CLI::GetSingleton().functionMap[d.tname]["AddToPO"](d, NULL,
-        (void*) &desc);
-
-    // Generate the name the user passes on the command line.
-    std::string boostName;
-    CLI::GetSingleton().functionMap[d.tname]["MapParameterName"](d, NULL,
-        (void*) &boostName);
-    boostNameMap[boostName] = d.name;
+    util::ParamData& d = it->second;
+    CLI::GetSingleton().functionMap[d.tname]["AddToPO"](d, NULL, (void*) &app);
```

Comment:
```suggestion
    IO::GetSingleton().functionMap[d.tname]["AddToPO"](d, NULL, (void*) &app);
```
Looks like the main compile issues past this point are just that the mlpack object `CLI` needs to be renamed in this directory (and maybe the other directories). :+1: